cli/proc_state.rs

```diff
@@ -204,6 +205,28 @@ impl ProcState {
         }
       };
 
+    if flags.compat {
+      let mut import_map = match maybe_import_map {
+        Some(import_map) => import_map,
+        None => {
+          // INFO: we're creating an empty import map, with its specifier pointing
+          // to `CWD/node_import_map.json` to make sure the map still works as expected.
+          let import_map_specifier =
+            std::env::current_dir()?.join("node_import_map.json");
+          ImportMap::from_json(import_map_specifier.to_str().unwrap(), "{}")
+            .unwrap()
+        }
+      };
+      let node_builtins = crate::compat::get_mapped_node_builtins();
+      let diagnostics = import_map.update_imports(node_builtins)?;
```

Comment:
What happens if some already has entries for some/all of them?

Reply:
A diagnostic is printed out to the terminal, like so:
```
❯ cargo run -- run --compat --unstable -A --import-map import-map.json cli/tests/testdata/compat/fs_promises.js
    Finished dev [unoptimized + debuginfo] target(s) in 0.38s
     Running `target/debug/deno run --compat --unstable -A --import-map import-map.json cli/tests/testdata/compat/fs_promises.js`
  - "fs" already exists and is mapped to Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("example.com")), port: None, path: "/1", query: None, fragment: None })
```
I just noticed that the output is broken and will fix it before landing the PR, should be
```
  - "fs" already exists and is mapped to "https://example.com/1"
```

Reply:
Added a test: `compat::existing_import_map`

Reply:
Finally shows up as:
```
The were problems adding Node built-ins to import map:
  - "fs/promises" already exists and is mapped to "[WILDCARD]non_existent_file.js"
```

Suggestions welcome

Reply:
I guess it is still unclear to the user what the state is or what action they should take. Is it really a problem? If it is a problem, what should a user do about it to resolve it?

Reply:
I clarified it even further:
```
Some Node built-ins were not added to the import map:
  - "fs/promises" already exists and is mapped to "[WILDCARD]non_existent_file.js"
If you want to use Node built-ins provided by Deno remove listed specifiers from "imports" mapping in the import map file.
```

Reply:
Cool